erpnext/selling/doctype/sales_order/sales_order.py

```diff
@@ -1257,19 +1257,50 @@ def create_muliple_production_plans(orders):
 
 @frappe.whitelist()
 def get_customer_item_ref_code(item, customer_name):
-    """Fetch the Customer Item Code for the given Item.
-    
-    Args:
-        item (varchar) : Item Code for the Sales Item 
-        customer_name (varchar) : Customer Name Of Sales Order
-
-    Returns:
-        Customer Item Code (varchar) : Returns the Customer Item Reference Code
-    """  	
-    customer_names = frappe.get_all("Item Customer Detail", filters={
-        "parent": item,
-        "customer_name": customer_name
-    }, fields=["ref_code"])    	
-
-    if customer_names:
-        return customer_names[0].ref_code
\ No newline at end of file
+	"""Fetch the Customer Item Code for the given Item.
+
+	Args:
+		item (varchar) : Item Code for the Sales Item
+		customer_name (varchar) : Customer Name Of Sales Order
+
+	Returns:
+		Customer Item Code (varchar) : Returns the Customer Item Reference Code
+	"""
+	customer_names = frappe.get_all("Item Customer Detail", filters={
+		"parent": item,
+		"customer_name": customer_name
+	}, fields=["ref_code"])
+
+	if customer_names:
+		return customer_names[0].ref_code
+
+@frappe.whitelist()
+def make_sales_order(source_name, target_doc=None):
```

Comment:
please rename the function to make_sales_order_from_batch if this is creating sales order only from a batch. Otherwise name is too generic